File: date_utc/csharp_date_utc/date_utc.cs
using System;

public class DateUTC {
  public static void Main( string[] args ) {
    if( args.Length != 0 ) {
      Console.WriteLine( "Usage: " + Environment.GetCommandLineArgs()[0] + " <noargs>" );
      Environment.Exit(1);
    }

    date_utc();
    date_today();
  }


  public static void date_today() {
    int year = DateTime.Now.Year;
    int month = DateTime.Now.Month;
    int day = DateTime.Now.Day;
    int hour = DateTime.Now.Hour;
    int min = DateTime.Now.Minute;
    int sec = DateTime.Now.Second;

    Console.WriteLine("utc=" + DateTime.Now.ToUniversalTime().ToString("yyyy-MM-dd HH:mm"));
    Console.WriteLine("local=" + DateTime.Now.ToString("yyyy-MM-dd  HH:mm"));
    Console.WriteLine("local dow=" + DateTime.Now.DayOfWeek);

    DateTime today = new DateTime(year, month, day, hour, min, sec);
    Console.WriteLine("create date=" + today);
    Console.WriteLine("local=" + year + "-" + month + "-" + day + " " + hour + ":" + min + ":" + sec);
  }

  public static void date_utc() {
    int year = DateTime.UtcNow.Year;
    int month = DateTime.UtcNow.Month;
    int day = DateTime.UtcNow.Day;
    int hour = DateTime.UtcNow.Hour;
    int min = DateTime.UtcNow.Minute;
    int sec = DateTime.UtcNow.Second;

    Console.WriteLine("utc" + DateTime.Now.ToUniversalTime().ToString("yyyy-MM-dd HH:mm"));
    Console.WriteLine("local=" + DateTime.Now.ToString("yyyy-MM-dd  HH:mm"));
    Console.WriteLine("utc dow=" + DateTime.UtcNow.DayOfWeek);

    DateTime today = new DateTime(year, month, day, hour, min, sec);
    Console.WriteLine("create date=" + today);
    Console.WriteLine("utc=" + year + "-" + month + "-" + day + " " + hour + ":" + min + ":" + sec);
  }
}

File: date_utc/python_date_utc/date_utc.py
import sys
import os
import time
import datetime
from datetime import datetime

def date_today_utc():
  utc = datetime.utcnow()
  print("utc: " + str(utc.year) + "-" + str(utc.month) + "-" + str(utc.day) + " " + str(utc.hour) + ":" + str(utc.minute) + ":" + str(utc.second))

def date_today():
  local = datetime.now()
  print("local: " + str(local.year) + "-" + str(local.month) + "-" + str(local.day) + " " + str(local.hour) + ":" + str(local.minute) + ":" + str(local.second))
  #print(local.strftime("%Y-%m-%d"))
  #print(local.strftime("%Y-%m-%d"))

def driver():
  time.time()
  #1226392970.4683549
  #time.gmtime(time.time()).tm_mday
  #time.gmtime(time.time()).tm_mon
  #time.gmtime(time.time()).tm_year
  x = time.gmtime(time.time())
  print(time.gmtime(time.time()).tm_mon )
  print(time.gmtime(time.time()).tm_mday)
  print(time.gmtime(time.time()).tm_year)
  print(time.gmtime(time.time()).tm_hour)
  print(time.gmtime(time.time()).tm_min )
  print(time.gmtime(time.time()).tm_sec )
  print(datetime.datetime.now())
  print("now: " ,datetime.datetime.now().strftime("%y-%m-%d %H:%M"))
#  print datetime.now().strftime("%Y-%m-%d %H:%M")

def main():
  if len(sys.argv) != 1:
    print("Usage: %s <noargs>" % sys.argv[0])
    sys.exit()
  #print(sys.argv[1])
  #driver()
  date_today_utc()
  date_today()
  sys.exit(), end

main()
